state/modelconfig.go

```diff
@@ -134,8 +135,18 @@ func (st *State) modelConfigValues(modelCfg attrValues) (config.ConfigValues, er
 }
 
 // UpdateModelConfigDefaultValues updates the inherited settings used when creating a new model.
-func (st *State) UpdateModelConfigDefaultValues(attrs map[string]interface{}, removed []string) error {
-	settings, err := readSettings(st, globalSettingsC, controllerInheritedSettingsGlobalKey)
+func (st *State) UpdateModelConfigDefaultValues(attrs map[string]interface{}, removed []string, cloudTag, regionName string) error {
+	var key string
+	if cloudTag != "" && regionName != "" {
+		cloud, err := names.ParseCloudTag(cloudTag)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		key = regionSettingsGlobalKey(cloud.Id(), regionName)
+	} else {
```

Comment:
Changed the machinery to use environs.RegionSpec to define the cloud/region and added a constructor that barfs if cloud or region is an empty string.